includes/admin/class-amp-story-templates.php

```diff
@@ -0,0 +1,197 @@
+<?php
+/**
+ * AMP Story Templates handler.
+ *
+ * @package AMP
+ * @since 1.?
+ */
+
+/**
+ * Class AMP_Story_Templates
+ */
+class AMP_Story_Templates {
+
+	/**
+	 * Current version of the templates.
+	 *
+	 * @var string
+	 */
+	const STORY_TEMPLATES_VERSION = '0.1';
+
+	/**
+	 * Slug for templates' taxonomy.
+	 */
+	const TEMPLATES_TAXONOMY = 'amp_template';
+
+	/**
+	 * Slug for templates term.
+	 */
+	const TEMPLATES_TERM = 'story-template';
+
+	/**
+	 * Init.
+	 */
+	public function init() {
+		if ( ! post_type_exists( 'amp_story' ) ) {
+			return;
+		}
+
+		add_filter( 'rest_wp_block_query', array( $this, 'filter_rest_wp_block_query' ), 10, 2 );
+
+		$this->register_taxonomy();
+		$this->maybe_import_story_templates();
+	}
+
+	/**
+	 * Import story templates if they haven't been imported previously.
+	 */
+	public function maybe_import_story_templates() {
+		if ( self::STORY_TEMPLATES_VERSION === AMP_Options_Manager::get_option( 'story_templates_version' ) ) {
+			return;
+		}
+		$this->import_story_templates();
+		// @todo This is commented out for testing purposes until the PR gets ready.
+		// AMP_Options_Manager::update_option( 'story_templates_version', self::STORY_TEMPLATES_VERSION );
+	}
+
+	/**
+	 * Import logic for story templates.
+	 */
+	public function import_story_templates() {
+		$templates = self::get_story_templates();
+		foreach ( $templates as $template ) {
+			$existing_template = $this->template_exists( $template['title'], $template['content'] );
+			if ( $existing_template > 0 ) {
+				continue;
+			}
+			$post_id = wp_insert_post(
+				array(
+					'post_title'   => $template['title'],
+					'post_type'    => 'wp_block',
+					'post_status'  => 'publish',
+					'post_content' => $template['content'],
+				)
+			);
+			wp_set_object_terms( $post_id, self::TEMPLATES_TERM, self::TEMPLATES_TAXONOMY );
+		}
+	}
+
+	/**
+	 * Get the static list of templates, including temlate name and content.
+	 *
+	 * @return array Story templates.
+	 */
+	public static function get_story_templates() {
+		return array(
+			array(
+				'title'   => __( 'Default Template 1', 'amp' ),
+				'content' => '<!-- wp:amp/amp-story-page {"backgroundColor":"#313131","autoAdvanceAfterDuration":0} -->
```

Comment:
Instead of putting the raw HTML into the PHP, how about loading the HTML from files on disk? This will make them easier to maintain.